kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionCoordinator.java

```diff
@@ -492,8 +492,17 @@ public void handleAddPartitionsToTransaction(String transactionalId,
                 return Either.left(producerEpochFenceErrors());
             } else if (txnMetadata.getPendingState().isPresent()) {
                 // return a retriable exception to let the client backoff and retry
+                if (log.isDebugEnabled()) {
+                    log.debug("Producer {} is in pending state {}, responding CONCURRENT_TRANSACTIONS",
+                            transactionalId, txnMetadata.getPendingState());
+                }
                 return Either.left(Errors.CONCURRENT_TRANSACTIONS);
             } else if (txnMetadata.getState() == PREPARE_COMMIT || txnMetadata.getState() == PREPARE_ABORT) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Producer {} is in state {}, responding CONCURRENT_TRANSACTIONS",
+                            transactionalId, txnMetadata.getState()
+                    );
+                }
```

Comment:
For the same reason as I added, you can hardly see such logs in Kafka.

Reply:
Adding this debug log will help to debug since they all return `Errors.CONCURRENT_TRANSACTIONS`.